parsers/gdscript.c

```diff
@@ -0,0 +1,789 @@
+
+		if (indent <= PY_NL_INDENTATION(nl))
+			break;
+		if (prev)
+		{
+			vStringPut(result, '.');	/* make Geany symbol list grouping work properly */
+/*
+			if (prev->kindIndex == K_CLASS)
+				vStringPut(result, ".");
+			else
+				vStringPut(result, "/");
+*/
+		}
+
+		e = getEntryOfNestingLevel (nl);
+		if (e)
+		{
+			vStringCatS(result, e->name);
+			is_class = (e->kindIndex == K_CLASS);
+		}
+		else
+			is_class = false;
+
+		prev = nl;
+	}
+	return is_class;
+}
+
+/* Check indentation level and truncate nesting levels accordingly */
+static void checkIndent(NestingLevels *nls, int indent)
+{
+	int i;
+	NestingLevel *n;
+
+	for (i = 0; i < nls->n; i++)
+	{
+		n = nestingLevelsGetNthFromRoot (nls, i);
+		if (n && indent <= PY_NL_INDENTATION(n))
+		{
+			/* truncate levels */
+			nls->n = i;
+			break;
+		}
+	}
+}
+
+static void addNestingLevel(NestingLevels *nls, int indentation, struct corkInfo *info)
+{
+	int i;
+	NestingLevel *nl = NULL;
+
+	for (i = 0; i < nls->n; i++)
+	{
+		nl = nestingLevelsGetNthFromRoot(nls, i);
+		if (indentation <= PY_NL_INDENTATION(nl)) break;
+	}
+	if (i == nls->n)
+		nl = nestingLevelsPush(nls, info->index);
+	else
+		/* reuse existing slot */
+		nl = nestingLevelsTruncate (nls, i + 1, info->index);
+
+	PY_NL_INDENTATION(nl) = indentation;
+}
+
+/* Return a pointer to the start of the next triple string, or NULL. Store
+ * the kind of triple string in "which" if the return is not NULL.
+ */
+static char const *find_triple_start(char const *string, char const **which)
+{
+	char const *cp = string;
+
+	for (; *cp; cp++)
+	{
+		if (*cp == '#')
+			break;
+		if (*cp == '"' || *cp == '\'')
+		{
+			if (strncmp(cp, doubletriple, 3) == 0)
+			{
+				*which = doubletriple;
+				return cp;
+			}
+			if (strncmp(cp, singletriple, 3) == 0)
+			{
+				*which = singletriple;
+				return cp;
+			}
+			cp = skipString(cp);
+			if (!*cp) break;
+			cp--; /* avoid jumping over the character after a skipped string */
+		}
+	}
+	return NULL;
+}
+
+/* Find the end of a triple string as pointed to by "which", and update "which"
+ * with any other triple strings following in the given string.
+ */
+static void find_triple_end(char const *string, char const **which)
+{
+	char const *s = string;
+	while (1)
+	{
+		/* Check if the string ends in the same line. */
+		s = strstr (s, *which);
+		if (!s) break;
+		s += 3;
+		*which = NULL;
+		/* If yes, check if another one starts in the same line. */
+		s = find_triple_start(s, which);
+		if (!s) break;
+		s += 3;
+	}
+}
+
+// TODO: Add typing and annotations
+static const char *findVariable(const char *line)
+{
+	/* Parse global and class variable names (C.x) from assignment statements.
+	 * Object attributes (obj.x) are ignored. */
+	const char *cp, *sp, *eq, *start, *ln;
+	
+	ln = line;
+	if (*ln == '@') {
+		++ln;
+		while (isIdentifierCharacter ((int) *ln))
+			++ln;
+		while (isspace ((int) *ln))
+			++ln;
+	}
+
+	if (strncmp (ln, "var", 3) != 0)
+		return NULL;
+
```

Comment:
Though it is invisible on the GitHub web interface, there are unneeded whitespaces. Could you remove them?